src/many-abci/src/abci_app.rs

```diff
@@ -172,6 +220,31 @@ impl Application for AbciApp {
         ResponseBeginBlock { events: vec![] }
     }
 
+    fn check_tx(&self, request: RequestCheckTx) -> ResponseCheckTx {
+        let (transmitter, receiver) = channel::<Option<transaction_cache::Value>>();
+        self.transmitter
+            .send(transaction_cache::Message::Get(
+                {
+                    let mut hasher = Sha256::new();
+                    hasher.update(request.tx);
+                    hasher.finalize().to_vec().into()
+                },
+                transmitter,
+            ))
+            .unwrap_or_default();
+        match receiver.recv() {
```

Comment:
And then when we see a new transaction we look for an existing transaction that's the same. If we can't find one, we allow the new transaction through?

Reply:
Yes, that's correct.